src/Entity/AuditRecord.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Entity;
+
+use App\Audit\AuditRecordType;
+use DateTimeImmutable;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\Selectable;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Query\Expr\Select;
+use Symfony\Component\Uid\Ulid;
+
+#[ORM\Entity(repositoryClass: AuditRecordRepository::class)]
+#[ORM\Table(name: 'audit_log')]
+#[ORM\Index(name: 'type_idx', columns: ['type'])]
+#[ORM\Index(name: 'datetime_idx', columns: ['datetime'])]
+#[ORM\Index(name: 'vendor_idx', columns: ['vendor'])]
+#[ORM\Index(name: 'package_idx', columns: ['packageId'])]
+class AuditRecord
+{
+    #[ORM\Id]
+    #[ORM\Column(type: 'ulid')]
+    public readonly Ulid $id;
+
+    #[ORM\Column(type: Types::DATETIME_IMMUTABLE)]
+    public readonly DateTimeImmutable $datetime;
+
+    private function __construct(
+        #[ORM\Column(length: 255, enumType: AuditRecordType::class)]
```

Comment:
I'm quite sure `enumType` is automatically guessed from the property type already.